src/modal/alert-modal.component.ts

```diff
@@ -0,0 +1,138 @@
+import {
+	Component,
+	Injector,
+	OnInit,
+	ElementRef,
+	AfterViewInit
+} from "@angular/core";
+import {
+	trigger,
+	state,
+	style,
+	transition,
+	animate
+} from "@angular/animations";
+import Modal from "./modal.decorator";
+import { ModalService } from "./modal.service";
+
+/**
+ * Component to create standard modals for presenting content or asking for user's input.
+ * It can show as a passive modal showing only text or show as a transactional modal with
+ * multiple buttons for different actions for the user to choose from.
+ *
+ * Using a modal in your application requires `ibm-modal-placeholder` which would generally be
+ * placed near the end of your app component template (app.component.ts or app.component.html) as:
+ *
+ * ```html
+ * <ibm-modal-placeholder></ibm-modal-placeholder>
+ * ```
+ *
+ * Example of opening the modal:
+ *
+ * ```typescript
+ * \@Component({
+ *  selector: "app-modal-demo",
+ *  template: `
+ *   <button class="btn--primary" (click)="openModal()">Open modal</button>
+ *   <ibm-modal-placeholder></ibm-modal-placeholder>`
+ * })
+ * export class ModalDemo {
+ *
+ * 	openModal() {
+ * 		this.modalService.show({
+ *			modalType: "default" | "danger",
+ *			headerLabel: "optional header text",
+ *			title: "Modal title",
+ *			text: "Modal text",
+ *			buttons: [{
+ *				text: "Button text",
+ *				type: "primary" | "secondary" | "tertiary" | "ghost" | "danger" | "danger--primary" = "primary",
+ *				click: clickFunction,
+ *			}]
+ *		});
+ * 	}
+ * }
+ * ```
+ *
+ * @export
+ * @class AlertModalComponent
+ */
+@Modal()
+@Component({
+	selector: "ibm-alert-modal",
+	template: `
+		<ibm-modal [modalType]="modalType">
+			<ibm-modal-header (closeSelect)="closeModal()">
+				<p class="bx--modal-header__label bx--type-delta">{{headerLabel}}</p>
+      			<p class="bx--modal-header__heading bx--type-beta">{{title}}</p>
+			</ibm-modal-header>
+			<div class="bx--modal-content">
+				<p>{{text}}</p>
+			</div>
+			<ibm-modal-footer *ngIf="buttons.length > 0">
+				<button ibmButton="{{button.type}}" *ngFor="let button of buttons; let i = index"
+					(click)="buttonClicked(i)" id="{{button.id}}">{{button.text}}</button>
```

Comment:
`id` binds cleaner if its `[id]="button.id"` ... also I'd stack those attributes like

```
<button
    ibmButton=""
   *ngFor
    etc>
    {{button.text}}
</button>
```

For consistency's sake